srv/modules/runners/proposal.py

```diff
@@ -308,6 +310,159 @@ def _record_filter(args, base_dir):
         yaml.dump(current_filter, filehandle, default_flow_style=False)
 
 
+def _find_minions_to_replace(profile_dir):
+    """
+    Search through a given profile directory for files that end with '-replace'.
+
+    :param profile_dir: Profile directory, e.g. "/srv/pillar/ceph/proposals/profile-default"
+    """
+    ToReplace = namedtuple('ToReplace', ['fullpath', 'filename'])
+    dir = '{}/stack/default/ceph/minions'.format(profile_dir)
+    files = [f for f in os.listdir(dir) if isfile(join(dir, f))]
```

Comment:
You could add the ```f.endswith('-replace')``` here and thus only iterate once over all profile files. Somehting like
```files = [f for f in os.listdir(dir) if isfile(join(dir, f)) and f.endswith('-replace')]```

Reply:
Good catch, that does not need two iterations over all profiles